bin/eslint-formatters-calypso/src/lib/differ.js

```diff
@@ -0,0 +1,83 @@
+/* eslint strict: "off" */
+
+'use strict';
+
+module.exports = function( diff ) {
+    /*
+    * This function returns a structure out of a git diff
+    * with the filenames and lines modified.
+    *
+    * returns {
+    * 'fileName': [1],
+    * 'fileName2': [2]
+    * }
+    *
+    */
+
+	const files = {};
+	if ( ! diff ) {
+		return files;
+	}
+
+	const lines = diff.split( '\n' );
+	let currentFile;
+	lines.forEach( line => {
+		if ( line.startsWith( 'diff --git' ) ) {
+			currentFile = getFile( line );
+			files[ currentFile ] = [];
+		} else if ( line.startsWith( '@@' ) ) {
+			getLineNumbers( line ).forEach( lineNumber => {
+				files[ currentFile ].push( lineNumber );
+			} );
+		}
+	} );
+
+	function getFile( strLine ) {
+        /*
+        *
+        * See https://git-scm.com/docs/git-diff
+        *
+        * ex: diff --git a/file1 b/file2diff
+        *
+        * The a/ and b/ filenames are the same unless rename/copy is involved.
+        * Especially, even for a creation or a deletion,
+        * /dev/null is not used in place of the a/ or b/ filenames.
+		*
+        * When rename/copy is involved, file1 and file2 show
+        * the name of the source file of the rename/copy and
+        * the name of the file that rename/copy produces, respectively.
+        *
+		* We expect a/ and b/ to be complete paths, which can be achieved
+		* by using --src-prefix and --dst-prefix.
+		*
+        */
+		return strLine.split( ' ' )[ 3 ];
+	}
+
+	function getLineNumbers( strLine ) {
+        /* eslint-disable */
+        // disable max-len rule due to long URI
+        /*
+        *
+        * See https://www.gnu.org/software/diffutils/manual/html_node/Detailed-Unified.html#Detailed-Unified
+        *
+        * ex: @@ -65,3 +65,3 @@ module.exports = {
+        *
+        * 65 is the starting point, 3 the number of lines affected,
```

Comment:
> 3 the number of lines affected

I think the `count`s for a hunk (the number after the comma) includes the context lines, not just the modified lines—see example below. So we will be including the context lines from the diff in the lint. Is that what we want? (Maybe it is.)

<img width="679" alt="screen shot 2016-07-27 at 22 22 20" src="https://cloud.githubusercontent.com/assets/7767559/17192920/8e99715a-5448-11e6-9164-0f681da1a1e3.png">